### DISCUSSION TITLE
Statistical test for competing events

### DISCUSSION THREAD

**Discussion by BorjaGIH** [CamDavidsonPilon/lifelines#1341] [Q&A] [answered]
Hi!

I am using the Aalen-Johansen estimator to model survival functions in presence of competing events. I would like to compare some curves using a statistical test. If I were using the Kaplan-Meier estimator (no competing events), I would use the log-rank test. 

My question is: is it correct to use that same test (log-rank) with the Aalen-Johansen estimator, with competing events?
The reasons for my doubts are: I read that Gray proposed an alternative test for this setting, based on a modified Chi-Square test statistic. So I assume it is not correct to use the log-rank test. But then I checked the log-rank test documentation from lifelines and I saw that it is using a Chi-Square test statistic at some point (https://lifelines.readthedocs.io/en/latest/Examples.html#statistically-compare-two-populations). All this confused me.

In addition: if this approach is not correct, does lifelines have an alternative? 

Thank you very much!

Borja

**Comment by pzivich** [comment, marked as answer]
Hi Borja,

The statistic in the log-rank comparison follows a Chi-squared distribution. So that is why the documentation talks about the Chi-square.

To compute the log-rank statistic, you need the number of events and the N at each time. However, I don't know if the log-rank still works as intended if you only consider the events of interest (and not the competing risks). I am unfamiliar with what Gray proposed as a modification (and you provide a link to the paper?). In my coursework on Aalen-Johansen, we did not talk about a comparison for those curves.

AFAIK lifelines doesn't currently have an option for inference between two curves in the Aalen-Johansen. But another way you could compare the curves is the calculate something like the difference between the two risks at each time. The variance can be estimated by adding the two variances together (via a delta-method argument). You can then plot the 1-\alpha confidence intervals. However, this approach does not test whether there is an overall difference (but I prefer it regardless because it provides more information).

**Comment by BorjaGIH** [comment]
Hello pzivich,

Thank you very much for your answer. This is the webpage where I learned about competing events framework, and about the modified test https://www.publichealth.columbia.edu/research/population-health-methods/competing-risk-analysis. I find the whole text pretty instructive and well explained. You can find the original paper here https://www.jstor.org/stable/pdf/2241622.pdf. Digesting the paper is not that simple, or at least requires more time. The webpage also gives some indications about a proportional hazards model for the competing event framework, very similar to Cox's. They call it Cumulative Incidence Function (CIF) proportional hazards model.

Regarding my particular problem, I may conduct a sensitivity analysis and see how much the KM and AJ estimators differ, and if the deviation is small, I may as well continue with my analysis assuming no competing risks.

Thank you.

**Reply by pzivich** [reply]
Thanks! I will have to read Gray 1998 to see how the test statistic is constructed.

Section 3.3 refers to the Fine-Gray model, which is related to the standard Cox PH model (but extended for competing risks). However, the website is a little misleading, since the Fine-Gray model is not parametric (it is semiparametric). That is a little different than the Aalen-Johansen estimator. The Fine-Gray model provides estimates of the cause-specific hazard ratio, unlike Aalen-Johansen which estimates the risk function.

As for using the Kaplan-Meier, if you have relatively few competing events that happen (something like 5), there won't be much of a difference between the two. Anything above that is likely going to cause differences. I would still recommend plotting the difference between the two estimated risks and their associated confidence intervals. IMO, that is more informative than a singular test statistic / p-value for a difference across the whole curves.

**Reply by BorjaGIH** [reply]
Thanks a lot for all the comments and the guidance. I have only 2 competing events, so I guess I will continue with the "simplified" analysis (KM), after conducting a sensitivity analysis. As for the test statistic thing, I will definitely plot the risk function (in my case the survival function) with the CIs.

I will mark your answer as selected answer. Thanks again.

Borja

**Comment by evannemartin** [comment]
Hello @pzivich ! I discovered this issue while looking for a Gray test in lifelines.
I assume it was not implemented in the meantime. It might be a good idea as I know that statisticians often use this test in SAS ?
